stl/inc/format

```diff
@@ -976,19 +976,21 @@ template <class _Context, class _Ty>
     }
 }
 
-template <class _Context, class _Char_type = typename _Context::char_type>
-/* consteval */ constexpr auto _Get_format_arg_storage_type(const _Char_type*) noexcept {
-    return static_cast<const _Char_type*>(nullptr);
+template <class _Context>
+/* consteval */ constexpr auto _Get_format_arg_storage_type(const typename _Context::char_type*) noexcept {
+    return static_cast<const typename _Context::char_type*>(nullptr);
 }
 
-template <class _Context, class _Traits, class _Char_type = typename _Context::char_type>
-/* consteval */ constexpr auto _Get_format_arg_storage_type(basic_string_view<_Char_type, _Traits>) noexcept {
-    return basic_string_view<_Char_type>{};
+template <class _Context, class _Traits>
+/* consteval */ constexpr auto _Get_format_arg_storage_type(
+    basic_string_view<typename _Context::char_type, _Traits>) noexcept {
+    return basic_string_view<typename _Context::char_type>{};

```

Comment:
Answering my own question: because the working draft says so. We ignore (instead of reject) non-default traits and simply allow the initialization of the exposition-only variant to be ill-formed and explode. That will produce _horrible_ error messages.

Reply:
Clarifying: the code already forces said conversion, but to my knowledge nobody has filed an LWG issue to allow that behavior.

Reply:
I'd be fine with adding a card to the Format Project as a reminder to file an LWG issue; shouldn't block merging this.

Reply:
I've recorded this in my list of issues; I'll file these today either here or with LWG.